src/pykeen/models/unimodal/auto_sf.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+"""A simple AutoSF-based model."""
+
+from typing import Any, ClassVar, Mapping, Optional, Sequence, Tuple
+
+from ..nbase import ERModel
+from ...constants import DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE
+from ...nn.emb import EmbeddingSpecification
+from ...nn.modules import AutoSFInteraction
+from ...typing import Sign
+
+__all__ = [
+    "AutoSF",
+]
+
+YAGO310_COEFFICIENTS = (
+    (0, 0, 0, 1),
+    (1, 1, 1, 1),
+    (1, 2, 3, -1),
+    (2, 2, 2, 1),
+    (3, 1, 1, -1),
+    (3, 4, 3, 1),
+)
+
+
+class AutoSF(ERModel):
+    r"""An implementation of AutoSF from [zhang2020]_.
+
+    ---
+    name: AutoSF
+    citation:
+        author: Zhang
+        year: 2020
+        arxiv: 1904.11682
+        link: https://arxiv.org/abs/1904.11682
+        github: AutoML-Research/AutoSF
+    """
+
+    #: The default strategy for optimizing the model's hyper-parameters
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        embedding_dim=DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE,
+    )
+
+    def __init__(
+        self,
+        embedding_dim: int = 256,
+        num_components: int = 4,

```

Comment:
is this necessary? can it be just counted as `len(coefficients)`?

Reply:
Yes, it is. `len(coefficients)` can be up to `num_components**3`. The coefficients describe which head/relation/tail component get combined with each other.

Reply:
🎶  update the docs plzzz 🎵